tests/test_scripts.py

```diff
@@ -69,12 +78,92 @@ def test_bulk_process(tmp_path):
     assert p.returncode == 0
 
 
+@pytest.mark.parametrize(
+    "script,args,expected_output_path",
+    [
+        # Test match-2-yar x86 EXE
+        pytest.param(
+            "match-2-yar.py",
+            [get_data_path("9324d1a8ae37a36ae560c37448c9705a.exe_")],
+            "yara/expected_9324d1a8ae37a36ae560c37448c9705a.exe_.yar",
+        ),
+        # Test match-2-yar x64 EXE
+        pytest.param(
+            "match-2-yar.py",
+            [get_data_path("c2bb17c12975ea61ff43a71afd9c3ff111d018af161859abae0bdb0b3dae98f9.exe_")],
+            "yara/expected_c2bb17c12975e.yar",
+        ),
+        # Test match-2-yar x86 .NET EXE
+        pytest.param(
+            "match-2-yar.py",
+            [
+                "-f",
+                "dotnet",
+                get_data_path("dotnet/1c444ebeba24dcba8628b7dfe5fec7c6.exe_"),
+            ],
+            "yara/expected_1c444ebeba24dcba8628b7dfe5fec7c6.exe_.yar",
+        ),
+        # Test match-2-yar files with multiple X86 PEs
+        pytest.param(
+            "match-2-yar.py",
+            [
+                get_data_path("Practical Malware Analysis Lab 03-04.exe_"),
+                get_data_path("Practical Malware Analysis Lab 11-03.exe_"),
+                get_data_path("Practical Malware Analysis Lab 16-01.exe_"),
+            ],
+            "yara/expected_pma_03-04.exe_11-03.exe_16-01.exe",
+        ),
+        # Test match-2-yar files with CAPA file limitations are filtered out of multi sample
+        pytest.param(
+            "match-2-yar.py",
+            [
+                get_data_path("Practical Malware Analysis Lab 01-01.exe_"),
+                get_data_path("Practical Malware Analysis Lab 01-02.exe_"),
+            ],
+            "yara/expected_pma_01-01.exe_01-02.exe",
+        ),
+        # Test match-2-yar multiple x86 .NET PE
+        pytest.param(
+            "match-2-yar.py",
+            [
+                "-f",
+                "dotnet",
+                get_data_path("dotnet/1c444ebeba24dcba8628b7dfe5fec7c6.exe_"),
+                get_data_path("dotnet/692f7fd6d198e804d6af98eb9e390d61.exe_"),
+            ],
+            "yara/expected_1c444ebe_692f7fd6.yar",
+        ),
+    ],
+)
+def test_script_expected_output(script, args, expected_output_path):
```

Comment:
I'm not convinced these are helpful tests down the road. Some files are very large and I doubt one would want to check them manually. I'm in favor of testing individual smaller functionality like done below.

Reply:
I agree you wouldn't be able to check the correctness of each yara rule manually but I do think they offer value as integration tests. 

The yara rules are designed to be deterministic. During my testing they did help identify some tricky sources of nondeterminism in comments for functions. 

Just having them as a sort of bellwether for changes I think is helpful, and the smaller unit tests will help hone in on specific issues when they arise. Potential disagreements between the two may even lead to better tests in the future

Reply:
ok, let's keep them then